proposals/025-lambda-1.md

```diff
@@ -0,0 +1,224 @@
+# SP #025: Lambda Expressions (Immutable Capture)
+
+This proposal adds initial support for lambda expressions in Slang.
+The initial proposal is to support immutable capture of variables in the surrounding scope.
+This means that the lambda can only read the values of the captured variables, not modify them.
```

Comment:
If there is a functional or language interaction reason for this restriction it would be nice to explain it. Or if it's just to reduce the scope of the initial implementation, that's also good to know.

Reply:
There is just to keep the initial scope small. There isn't a fundamental reason why we can't allow mutable captures, especially when targeting SPIRV/Metal where pointers are available.